src/components/Development/Sentry.tsx

```diff
@@ -1,65 +1,71 @@
 
-import {
-  BrowserOptions,
-  BrowserTracing,
-  Replay,
-  init,
-  setExtra,
-} from '@sentry/browser'
+import { BrowserOptions } from '@sentry/browser'
 
 import env from '../../utils/env'
 
-export default React.memo(function Sentry({ dsn, ...props }: BrowserOptions) {
-  dsn = dsn || env('SENTRY_DSN', process.env.GATSBY_SENTRY_DSN || '')
+type SentryProps = {
+  src?: string
+} & React.ScriptHTMLAttributes<HTMLScriptElement> &
+  React.HTMLProps<HTMLScriptElement>
 
-  if (!dsn) {
+export default React.memo(function Sentry({ src, ...props }: SentryProps) {
+  src = src || env('SENTRY_SRC', process.env.GATSBY_SENTRY_SRC || '')
+
+  if (!src) {
     console.warn(
-      `skipping Sentry init inject: dsn and GATSBY_SENTRY_DSN is missing'`
+      `skipping Sentry init inject: src and GATSBY_SENTRY_SRC is missing'`
     )
     return null
   }
 
+  const tags: Record<string, string | number> = {}
+
   const COMMIT_SHA = env('COMMIT_SHA')
   if (COMMIT_SHA !== undefined) {
-    setExtra('COMMIT_SHA', COMMIT_SHA)
+    tags['commit.sha'] = COMMIT_SHA
   }
 
   const COMMIT_SHORT_SHA = env('COMMIT_SHORT_SHA')
   if (COMMIT_SHORT_SHA !== undefined) {
-    setExtra('COMMIT_SHORT_SHA', COMMIT_SHORT_SHA)
+    tags['commit.short_sha'] = COMMIT_SHORT_SHA
   }
 
   const COMMIT_REF_NAME = env('COMMIT_REF_NAME')
   if (COMMIT_REF_NAME !== undefined) {
-    setExtra('COMMIT_REF_NAME', COMMIT_REF_NAME)
+    tags['commit.ref_name'] = COMMIT_REF_NAME
   }
 
   const COMMIT_BRANCH = env('COMMIT_BRANCH')
   if (COMMIT_BRANCH !== undefined) {
-    setExtra('COMMIT_BRANCH', COMMIT_BRANCH)
+    tags['commit.branch'] = COMMIT_BRANCH
   }
 
   const COMMIT_TAG = env('COMMIT_TAG')
   if (COMMIT_TAG !== undefined) {
-    setExtra('COMMIT_TAG', COMMIT_TAG)
+    tags['commit.tag'] = COMMIT_TAG
   }
 
-  useEffect(() => {
-    if (!dsn) return
-
-    init({
-      environment: env('ENVIRONMENT', 'local'),
-      dsn,
-      integrations: [new BrowserTracing(), new Replay()],
-      // Performance Monitoring
-      tracesSampleRate: 0.001, // Capture 1% of the transactions
-      // Session Replay
-      replaysSessionSampleRate: 0.1, // This sets the sample rate at 10%. You may want to change it to 100% while in development and then sample at a lower rate in production.
-      replaysOnErrorSampleRate: 1.0, // If you're not already sampling the entire session, change the sample rate to 100% when sampling sessions where errors occur.
-      ...props,
-    })
-  }, [dsn])
+  const initialConfig: BrowserOptions = {
+    environment: env('ENVIRONMENT', 'local'),
+    // Performance Monitoring
+    tracesSampleRate: 0.001,
+    // Session Replay
+    replaysSessionSampleRate: 0.1,
+    replaysOnErrorSampleRate: 1.0,
+  }
 
-  return <></>
+  return (
+    <>
+      <script {...props} src={src} crossOrigin="anonymous" />
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `Sentry.onLoad(function() {
+            Sentry.init(${JSON.stringify(initialConfig)})
```

Comment:
Using string concatenation to construct JavaScript code can be error-prone, or in the worst case, enable code injection if an input is constructed by an attacker.

## Recommendation

If using JSON.stringify or an HTML sanitizer to sanitize a string inserted into JavaScript code, then make sure to perform additional sanitization or remove potentially dangerous characters.

Please follow the `Intercom` component as an example to leverage over the React sanitization:  https://github.com/decentraland/decentraland-gatsby/blob/master/src/components/Development/Intercom.tsx

## References

OWASP: [Code Injection](https://www.owasp.org/index.php/Code_Injection).
Common Weakness Enumeration: [CWE-94](https://cwe.mitre.org/data/definitions/94.html).
Common Weakness Enumeration: [CWE-79](https://cwe.mitre.org/data/definitions/79.html).
Common Weakness Enumeration: [CWE-116](https://cwe.mitre.org/data/definitions/116.html).

Reply:
What about now @2fd ?

Reply:
I chose this option because I tried to init the lib in a `useEffect` that wasn't executed because the Sentry component was server-side rendered (using `gatsby-ssr`).

What should be the right solution here? Would be okay to render the component in the `gatsby-browser` file?